pkg/disk/x509_test.go

```diff
@@ -0,0 +1,142 @@
+
+	tempDir := t.TempDir()
+
+	tests := []struct {
+		name                    string
+		ca                      *spiffetest.CA
+		federatedCA             *spiffetest.CA
+		spiffeIDString          string
+		chainLength             int
+		includeFederatedDomains bool
+		intermediateInBundle    bool
+	}{
+		{
+			name:                    "Single SVID",
+			ca:                      rootCA,
+			spiffeIDString:          "spiffe://example.test/workload",
+			chainLength:             1,
+			includeFederatedDomains: false,
+			intermediateInBundle:    false,
+		},
+		{
+			name:                    "Single SVID with intermediate in bundle",
+			ca:                      rootCA,
+			spiffeIDString:          "spiffe://example.test/workload2",
+			chainLength:             1,
+			includeFederatedDomains: false,
+			intermediateInBundle:    true,
+		},
+		{
+			name:                    "SVID with intermediate CA",
+			ca:                      intermediateCA,
+			spiffeIDString:          "spiffe://example.test/workloadWithIntermediate",
+			chainLength:             2,
+			includeFederatedDomains: false,
+			intermediateInBundle:    false,
+		},
+		{
+			name:                    "SVID with intermediate CA and intermediate in bundle",
+			ca:                      intermediateCA,
+			spiffeIDString:          "spiffe://example.test/workloadWithIntermediate",
+			chainLength:             2,
+			includeFederatedDomains: false,
+			intermediateInBundle:    true,
+		},
+		{
+			name:                    "SVID with federated trust domains",
+			ca:                      rootCA,
+			federatedCA:             federatedCA,
+			spiffeIDString:          "spiffe://example.test/workload",
+			chainLength:             1,
+			includeFederatedDomains: true,
+			intermediateInBundle:    false,
+		},
+		{
+			name:                    "SVID with federated trust domains and intermediate in bundle",
+			ca:                      intermediateCA,
+			federatedCA:             federatedCA,
+			spiffeIDString:          "spiffe://example.test/workload",
+			chainLength:             2,
+			includeFederatedDomains: true,
+			intermediateInBundle:    true,
+		},
+	}
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			spiffeID, err := spiffeid.FromString(test.spiffeIDString)
+			require.NoError(t, err)
+			certs, key := test.ca.CreateX509SVID(spiffeID.String())
+			require.Len(t, certs, test.chainLength)
+
+			x509Context := &workloadapi.X509Context{
+				Bundles: x509bundle.NewSet(x509bundle.FromX509Authorities(spiffeID.TrustDomain(), test.ca.Roots())),
+				SVIDs: []*x509svid.SVID{
+					{
+						ID:           spiffeID,
+						Certificates: certs,
+						PrivateKey:   key,
+					},
+				},
+			}
+
+			bundle := test.ca.Roots()
+
+			if test.intermediateInBundle {
+				bundle = append(bundle, certs[1:]...)
+				certs = certs[:1]
+			}
+
+			if test.includeFederatedDomains {
+				federatedSpiffeID, err := spiffeid.FromString("spiffe://federated.test/server")
```

Comment:
why not creating a spiffeid.TrustDomain instead? you dont need the spiffeid here

Reply:
good catch, done